diff -uNr -x target hadoop-3.2.1-src/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/lib/common/sasl_digest_md5.cc hadoop-3.2.1-src_moded/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/lib/common/sasl_digest_md5.cc
--- hadoop-3.2.1-src/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/lib/common/sasl_digest_md5.cc	2019-09-10 22:35:49.000000000 +0800
+++ hadoop-3.2.1-src_moded/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/lib/common/sasl_digest_md5.cc	2020-08-28 23:29:53.218043717 +0800
@@ -94,7 +94,7 @@
 void DigestMD5Authenticator::GenerateCNonce() {
   if (!TEST_mock_cnonce_) {
     char buf[8] = {0,};
-    RAND_pseudo_bytes(reinterpret_cast<unsigned char *>(buf), sizeof(buf));
+    RAND_bytes(reinterpret_cast<unsigned char *>(buf), sizeof(buf));
     cnonce_ = Base64Encode(std::string(buf, sizeof(buf)));
   }
 }
diff -uNr -x target hadoop-3.2.1-src/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/lib/common/util.cc hadoop-3.2.1-src_moded/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/lib/common/util.cc
--- hadoop-3.2.1-src/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/lib/common/util.cc	2019-09-10 22:35:49.000000000 +0800
+++ hadoop-3.2.1-src_moded/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/lib/common/util.cc	2020-08-28 23:30:05.794293608 +0800
@@ -75,7 +75,7 @@
 
 std::string GetRandomClientName() {
   std::vector<unsigned char>buf(8);
-  RAND_pseudo_bytes(&buf[0], 8);
+  RAND_bytes(&buf[0], 8);
 
   std::ostringstream oss;
   oss << "DFSClient_"  << getpid() <<  "_" <<
diff -uNr -x target hadoop-3.2.1-src/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfs-tests/CMakeLists.txt hadoop-3.2.1-src_moded/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfs-tests/CMakeLists.txt
--- hadoop-3.2.1-src/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfs-tests/CMakeLists.txt	2019-09-10 22:35:49.000000000 +0800
+++ hadoop-3.2.1-src_moded/hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfs-tests/CMakeLists.txt	2020-08-28 23:29:03.397089775 +0800
@@ -37,7 +37,7 @@
 )
 
 add_executable(test_native_mini_dfs test_native_mini_dfs.c)
-target_link_libraries(test_native_mini_dfs native_mini_dfs ${JAVA_JVM_LIBRARY})
+target_link_libraries(test_native_mini_dfs native_mini_dfs ${JAVA_JVM_LIBRARY} ${CMAKE_DL_LIBS})
 add_test(test_test_native_mini_dfs test_native_mini_dfs)
 
 add_executable(test_htable ../libhdfs/common/htable.c test_htable.c)
